runhouse/servers/obj_store.py

```diff
@@ -753,7 +754,212 @@ def rename(self, old_key: Any, new_key: Any):
+            # and need to store it to be retrieved later. In this case we return a "retrievable".
+            logger.debug(
+                f"{self.servlet_name} servlet: Method {method_name} on module {key} is a {laziness_type}. "
+                f"Storing result to be retrieved later at result key {res}."
+            )
+            fut = self.construct_call_retrievable(res, run_name, laziness_type)
+            self.put_local(run_name, fut)
+            return fut
+
+        if remote and isinstance(res, Module):
+            # If the result is a module, we return just the config
+            res.name = res.name or run_name
+            self.put_local(res.name, res)
+            config = res.config_for_rns
+            return config
+
+        return res
+
+    @staticmethod
+    def construct_call_retrievable(res, res_key, laziness_type):
+        if laziness_type == "coroutine":
+            from runhouse.resources.future_module import FutureModule
+
+            # TODO make this one-time-use
+            return FutureModule(future=res, name=res_key)
+
+        elif laziness_type == "generator":
+            from runhouse.resources.future_module import GeneratorModule
+
+            return GeneratorModule(future=res, name=res_key)
+
+        elif laziness_type == "async generator":
+            from runhouse.resources.future_module import AsyncGeneratorModule
+
+            return AsyncGeneratorModule(future=res, name=res_key)
+
+        else:
+            raise ValueError(f"Invalid laziness type {laziness_type}")
+
+    def call(
+        self,
+        key: str,
+        method_name: str,
+        data: Any = None,
+        serialization: Optional[str] = None,
+        run_name: Optional[str] = None,
+        stream_logs: bool = False,
+        remote: bool = False,
+    ):
+        env_servlet_name_containing_key = self.get_env_servlet_name_for_key(key)
+
+        if (
+            env_servlet_name_containing_key == self.servlet_name
+            and self.has_local_storage
+        ):
+            from runhouse.servers.http.http_utils import deserialize_data
+
+            args, kwargs = deserialize_data(data, serialization) if data else ([], {})
+
+            res = self.call_local(
+                key,
+                method_name,
+                run_name=run_name,
+                stream_logs=stream_logs,
+                remote=remote,
+                *args,
+                **kwargs,
+            )
+        else:
+            res = self.call_for_env_servlet_name(
+                env_servlet_name_containing_key,
+                key,
+                method_name,
+                data=data,
+                serialization=serialization,
+                run_name=run_name,
+                stream_logs=stream_logs,
+                remote=remote,
+            )
+
+        if remote and isinstance(res, dict):
+            config = res
+            if config["system"] == self.get_cluster_config():
```

Comment:
how would the `config["system"]` be a config and not an initialized cluster object? people would only set that up via `.to(rh.here)` or `(..., system=rh.here)` right?